tests/h/formatters/annotation_hidden_test.py

```diff
@@ -112,8 +118,13 @@ def moderation_svc(db_session):
 
 
 @pytest.fixture
-def moderator_check(moderated_group):
-    return lambda group: (group == moderated_group)
+def has_permission(moderated_group):
+    def has_permission(permission, context):
+        return context.group == moderated_group
+
+    mocked_has_permission = create_autospec(has_permission)
+    mocked_has_permission.side_effect = has_permission
+    return mocked_has_permission
```

Comment:
I think you might be able to save a couple of lines:

```suggestion
def has_permission(moderated_group):
    def has_permission(permission, context):
        return context.group == moderated_group

    return create_autospec(has_permission, side_effect=has_permission)
```

I haven't tried this but you might be able to autospec the real `has_permission` which is just going to do a better job of making sure that the mock stays in sync with the real thing:

```suggestion
def has_permission(moderated_group, pyramid_request):
    def has_permission(permission, context):
        return context.group == moderated_group

    return create_autospec(pyramid_request.has_permission, side_effect=has_permission)
```

There was already a mock here so that's fine, but just as a general comment I think it's usually best to avoid mocking Pyramid things like `has_permission()`. Instead you can use the [`pyramid.testing`](https://docs.pylonsproject.org/projects/pyramid/en/latest/narr/testing.html) stuff (e.g. `testing_securitypolicy()`) to set up situations and then pass in the real `request.has_permission`.

Reply:
Yeah that's a lot better, I'll give it a shot.

Reply:
Yep, it's fine.